src/Nethermind/Nethermind.Blockchain/Producers/BaseBlockProducer.cs

```diff
@@ -78,7 +81,7 @@ protected Task<bool> TryProduceNewBlock(CancellationToken token)
                 {
                     if (Logger.IsWarn) Logger.Warn($"Preparing new block - parent header is null");
                 }
-                else if (_sealer.CanSeal(parentHeader.Number + 1, parentHeader.Hash))
+                else if (_signer.CanSign && _sealer.CanSeal(parentHeader.Number + 1, parentHeader.Hash))
```

Comment:
this I do not get - the role of the sealer is to sign / seal the block - why would the base producer have two clasess - a signer and a sealer? definitely something wrong here

Reply:
Probably some of the sealers should have a signer internally and respond to CanSeal with false if they cannot